lib/nodes/decorators/While.ts

```diff
@@ -24,6 +24,11 @@ export class While<S extends BlueshellState, E> extends Decorator<S, E> {
 		super('While-' + desc, child);
 	}
 
+	// override getNodeStorage to explicitly return WhileNodeStorage so consumers don't need to cast types
+	override getNodeStorage(state: S): WhileNodeStorage {
```

Comment:
@asdo6rs What do you think about this concept? It allows classes that extend `While` to use `this.getNodeStorage` and let implicit typing take over so there's no need to import `WhileNodeStorage` necessarily. 

I'm worried that it's kind of weird for just this one node to implement this pattern without propagating it all throughout Blueshell. That said, if we do like this pattern then we can make some issues to go and implement it for all the relevant nodes. The pattern being override `getNodeStorage` whenever a node is extending/using a custom NodeStorage.

Reply:
`RetryNodeStorage` is the only other one I can see, we could just add it now.

Reply:
Hmm, the other way I could see of doing this is to add a 3rd type parameter to the generic that defaults to NodeStorage if not specified.

Reply:
That's interesting, that would work too and with a lot less code duplication. If we're going to do it at all then I think that's a good approach.

Reply:
I think for now we are going to just export the interfaces. We'd like to add a 3rd type parameter to the generic args but we need to take more time than we have now in order to be careful and get it right.